modules/ACL/ACLController.php

```diff
@@ -57,6 +57,17 @@ static function checkAccess($category, $action, $is_owner=false, $type='module',
 		if(is_admin($current_user))return true;
 		//calendar is a special case since it has 3 modules in it (calls, meetings, tasks)
 
+      if ($category === AOS_Products_Quotes::class) {
+          return ACLAction::userHasAccess(
+              $current_user->id,
+              AOS_Quotes::class,
```

Comment:
AOS_Products_Quotes is also used on invoices and contracts

Reply:
Yes I had not noticed this. I've updated the code, but I'm not sure if this is correct. If a user has access to Quotes but not Contracts for example, would it allow this user to access the line items for Contracts as well?

For the calendar module, just below this portion of code, there is the same kind of logic.